test/integration/targets/docker_container/tasks/tests/options.yml

```diff
@@ -2440,18 +2440,50 @@
     force_kill: yes
   register: network_mode_3
 
-- name: cleanup
+- name: network_mode (container mode setup)
+  docker_container:
+    image: alpine:3.8
+    command: '/bin/sh -c "sleep 10m"'
+    name: "{{ cname_h1 }}"
+    state: started
+
+- name: network_mode (container mode)
   docker_container:
+    image: alpine:3.8
+    command: '/bin/sh -c "sleep 10m"'
     name: "{{ cname }}"
+    state: started
+    network_mode: "container:{{ cname_h1 }}"
```

Comment:
You should always add `force_kill: yes` if you change something, otherwise docker will wait 10 seconds or so when stopping.